develop to trunk

sasangachathumal opened the pull request:
**Story/Ticket Reference ID from JIRA:**
`Add link here`

**Additional Comments:**
`Add here`

**Checklist:**

- [x] Unit test completed?: (Y/N)
- [x] Docs folder up to date?: (Y/N)
- [x] Add if anything missed: (Y/N)
- [x] When merging to trunk did you do a version bump in the package.json file? (Y/N)

All Checklist are okay.

This PR looks great - it's ready to merge! Please review and let's ship it. :)

sasangachathumal committed:
[CORE-4349] added sonar-project.properties file.

sasangachathumal committed:
[CORE-4361] remove tsLint and added esLint

sasangachathumal committed:
[CORE-4361] added lodash package.
update index.ts by adding construtor to pass api url.
create new function login in index.ts calling auth service.
created new auth auth service with login function.
added all auth endpoints to auth service.
created new request service to make get and post requests.
create usils service.
created function to create full api url.
added unit test to all new ts files and functions.

sasangachathumal committed:
[CORE-4261] fixed some sonar cloud issues.

shawnm0705 committed:
Merge pull request #1 from intersective/feature/CORE-4361/setup-way-to-handle-configuration-to-package

Feature/core 4361/setup way to handle configuration to package

sasangachathumal committed:
added method to call forgot password api.
update index.ts, auth-service.ts with new mentods.
update npm package publish script.
update read me files.

sasangachathumal committed:
[CORE-4367] added unit test.

sasangachathumal committed:
[CORE-4367] updated readme file.

sasangachathumal committed:
Merge pull request #2 from intersective/feature/CORE-4367/create-forgot-password-service-in-practera-sdk

Feature/core 4367/create forgot password service in practera sdk

sasangachathumal committed:
Update README.md

Fixed Typo in README.md

sasangachathumal committed:
[CORE-4368] create methods for reset password.
update auth-service.ts and index.ts with methods.
added put request to request service.
updated readme.md file.
added unit test for all new changes.

sasangachathumal committed:
Merge pull request #3 from intersective/feature/CORE-4368/create-reset-password-service-in-preactera-sdk

[CORE-4368] Create Reset Password Service in Preactera SDK Package

sasangachathumal committed:
[CORE-4388] updated readMe.md file.
create new methods for mfa services.
added new methods to index.ts.
added new method to auth-service.ts.
added unit test.
added comments.

sasangachathumal committed:
[CORE-4388] fixed sonar issue

shawnm0705 committed:
Merge pull request #4 from intersective/feature/CORE-4388/create-mfa-services-in-practera-sdk

[CORE-4388] Create MFA Services in Practera SDK

trtshen committed:
[CORE-4385] added registration sdk

trtshen committed:
[CORE-4385] allow the use of ts-jest on test scripts

trtshen committed:
[CORE-4385] lint fix

trtshen committed:
[CORE-4385] test working by mocking post call function

sasangachathumal committed:
[CORE-4386] Created new methods for get config.
Added new methods to auth service and index files.
Added unit test to cover new methods.
Updated the readme.md file with new info.

sasangachathumal committed:
[CORE-4386] fixed sonar code smells.

sasangachathumal committed:
Merge pull request #6 from intersective/feature/CORE-4387/create-get-custom-config-service-in-practera-sdk

Feature/core 4387/create get custom config service in practera sdk

trtshen committed:
[CORE-4385] added error handling

trtshen committed:
Merge branch 'develop' into feature/CORE-4385/registration-service

trtshen committed:
[CORE-4385]

trtshen committed:
Update registration-service.spec.ts

trtshen committed:
Update registration-service.spec.ts

trtshen committed:
[CORE-4429] apiKey

trtshen committed:
[CORE-4385]

trtshen committed:
[CORE-4385] index.ts coverage 100%

trtshen committed:
Merge branch 'feature/CORE-4385/registration-service' into feature/CORE-4429/accept-apikey-in-constructor

trtshen committed:
[CORE-4429] fixed broken test due to sonar implement

trtshen committed:
[CORE-4385] acommodate sonar

trtshen committed:
Update auth-service.spec.ts

trtshen committed:
Update jest.config.js

trtshen committed:
Update jest.config.js

trtshen committed:
Update index.spec.ts

trtshen committed:
Update auth-service.spec.ts

trtshen committed:
Update index.spec.ts

trtshen committed:
redundancy

trtshen committed:
Update jest.config.js

trtshen committed:
Update jest.config.js

trtshen committed:
Update lint_unitest-feature.yml

trtshen committed:
Update lint_unitest-develop.yml

trtshen committed:
Revert "Update lint_unitest-develop.yml"

This reverts commit 32b467857896357d125259d518d8260b442d3db7.

trtshen committed:
Revert "Update lint_unitest-feature.yml"

This reverts commit e548f06455b5d7a7cdc662758ad88329c1035402.

Sunilsbcloud committed:
fix sonar issue

sasangachathumal committed:
Merge pull request #7 from intersective/feature/CORE-4385/registration-service

[CORE-4385] Feature/registration service

trtshen committed:
[CORE-4429]

sasangachathumal committed:
Merge pull request #8 from intersective/feature/CORE-4429/accept-apikey-in-constructor

[CORE-4429]

trtshen committed:
[CORE-4428] readme + tests

trtshen committed:
Update utils-service.ts

sasangachathumal committed:
Merge pull request #9 from intersective/feature/CORE-4428/error-handling

[CORE-4428] Feature/error-handling-urlformatter

sasangachathumal committed:
removed package.lock.json from repo.

sasangachathumal committed:
Update readme file.
added table of content, helth to readme file.

shawnm0705 committed:
renaming

shawnm0705 commented on src/index.ts:
```diff
@@ -1,5 +1,219 @@
+   * }
+   * @returns promise
+   */
+  forgotPassword(data: any): Promise<any> {
```
Please put a type for the `data`

shawnm0705 commented on src/services/utils/utils-service.ts:
```diff
@@ -0,0 +1,38 @@
+ * @param endpoint string - Api endpoint
+ * @returns string - Full API URL.
+ */
+export const createFullApiUrl = (apiUrl: string, endPoint: string): string => {
```
Why do we have this function since it is just invoking another function?

shawnm0705 commented on src/services/auth/auth-service.ts:
```diff
@@ -0,0 +1,190 @@
+import { makeGetApiCall, makePostApiCall, makePutApiCall } from '../request/request-service';
+import { createFullApiUrl } from '../utils/utils-service';
+import _ from 'lodash';
```
Let's put the import of lodash in the utils file, so that if we wanna change it in the future, we can change it in one place

sasangachathumal committed:
Updated email to username in login service.

shawnm0705 commented on src/services/utils/utils-service.ts:
```diff
@@ -0,0 +1,38 @@
+/**
```
General suggestion. The term `service` is used in Angular. As for this SDK, there's no `component` `module`, so there's no need for the concept of `service`. We can just name them without the `-service` suffix.

For this `utils` and `request`, do we expect other files in the future to be in the same folder? If not, we'd better not create a folder for it, we can just put the file under `src` instead. Same with the `auth`, `registration` folder

sasangachathumal committed:
[CORE-4448] move lodash to util service.

sasangachathumal committed:
[CORE-4448] Update the file structure.
update test configaration.

sasangachathumal committed:
[CORE-4448]moved core-api call to core-api.ts
added types to methods in index.ts.

sasangachathumal committed:
[CORE-4448] pass multiple url through constructor.
create a method to update them.
use that same method inside constructor also.
added new error messages.
added logic to check api url from the methods it use.
added unit test to cover all that changes.

sasangachathumal committed:
[CORE-4448] Updated the read me file.

sasangachathumal committed:
[CORE-4448] removed createFullApiUrl from utils.
update api files to use urlFormatter.
update unit test related to that.
added same utint test to urlFormatter from cutie.

sasangachathumal committed:
Merge pull request #11 from intersective/feature/CORE-4448/code-refactoring-and-fixing-issues

Feature/core 4448/code refactoring and fixing issues

sasangachathumal committed:
[CORE-4448] renamed request methods.

sasangachathumal committed:
[CORE-4448] core-api, login-api convert to class. move all the validations to core-api.ts and login-api.ts. created more interfaces. make methods in login-api, core-api to update class parameters. comment unit test for now.

sasangachathumal committed:
[CORE-4448] created some mock files. added unit test for core-api.ts in automatic mock way.

sasangachathumal committed:
[CORE-4448] update the export to default export.

sasangachathumal committed:
[CORE-4448] Updated core-api.spec.ts to use mock.
update index.spec.ts to use ject mock.
updated login.spec.ts to use jest mock.
remove checks from constructor of core-api.ts.
remove checks from constructor of login-api.ts.

sasangachathumal committed:
Merge pull request #12 from intersective/feature/CORE-4448/code-refactoring-and-fixing-issues

[CORE-4448] renamed request methods.

sasangachathumal committed:
added direct login method to login-api.ts.
added unit test.
added new method to index.ts.
update index.ts to export default.

sasangachathumal committed:
Update readme.md

sasangachathumal committed:
Merge pull request #13 from intersective/feature/add-direct-login-service

Created Direct Login Service.
